rr_viz/src/rr_viz/widgets/mission_dashboard/path_record_panel.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+import os
+import rospy
+from PyQt5 import QtGui, QtWidgets, uic, QtCore
+from PyQt5.QtWidgets import QApplication, QWidget, QInputDialog, QLineEdit, QFileDialog, QMessageBox, QListWidgetItem
+from rr_2d_navigation.msg import RecordPathAction, RecordPathResult, RecordPathFeedback, RecordPathActionGoal
+from std_srvs.srv import Trigger
+from helpers import rr_qt_helper
+import actionlib
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+rr_viz_dir = _dir = os.path.dirname(os.path.dirname(current_dir))
+Form, Base = uic.loadUiType(os.path.join(current_dir, "path_record_panel.ui"))
+
+
+class PathRecordPanelWidget(Base, Form):
+    ''' The only reason this exists so that we can connect mission_editor to command_panel  easily '''
+    set_enabled = QtCore.pyqtSignal(bool)
+    def __init__(self, parent=None):
+        super(self.__class__, self).__init__(parent)
+        self.setupUi(self)
+        self.set_enabled.connect(self.setEnabled)
+        self.stopRecordingButton.setup(
+            "/path_recorder/finish_path", Trigger)
+        self.startRecordingButton.clicked.connect(self.startRecordingSlot)
+        self.statusLabel.setup("/path_recorder/status")
+        self._client = actionlib.SimpleActionClient('/path_recorder/record', RecordPathAction)
+        self.state_checker = rr_qt_helper.StateCheckerTimer(
+            self.is_record_up, self.set_enabled, Hz=1./3.)
+        self.state_checker.start()        
+
+    def is_record_up(self):        
+        return self._client.wait_for_server(timeout=rospy.Duration(2.))    
+
+    def startRecordingSlot(self):
+        goal=RecordPathActionGoal().goal        
+        goal.file_path=""
```

Comment:
I think this has to be set, otherwise I'm getting this error:

![image](https://user-images.githubusercontent.com/6543766/104322258-069b0a00-54e5-11eb-9f8e-4e1fc8d455ce.png)

Recording path to a file was requirement when I was working on the path follower. I'm open to the idea of not requiring saving it to the path, we we just need some direction from Pepe.

Reply:
I might not be getting this, but the question is if we have a default path file path or not? Kinda like make a path, and either save it or keep it in RAM to then follow?

Reply:
There is no file path set in the node but it can be set through a parameter. This path provided at launch will be used if the string used in the action call is empty.